test/mock/createInstantSearch.ts

```diff
@@ -2,26 +2,65 @@ import algoliasearchHelper from 'algoliasearch-helper';
 import index from '../../src/widgets/index/index';
 import { InstantSearch } from '../../src/types';
 import { createSearchClient } from './createSearchClient';
+import defer from '../../src/lib/utils/defer';
 
 export const createInstantSearch = (
   args: Partial<InstantSearch> = {}
 ): InstantSearch => {
   const searchClient = createSearchClient();
-  const mainHelper = algoliasearchHelper(searchClient, 'index_name', {});
-  const mainIndex = index({ indexName: 'index_name' });
+  const indexName = 'index_name';
+  const mainHelper = algoliasearchHelper(searchClient, indexName, {});
+  const mainIndex = index({ indexName });
+
+  let started = false;
 
   return {
+    indexName,
     mainIndex,
     mainHelper,
+    client: searchClient,
+    started,
+    start: () => {
+      started = true;
+    },
+    dispose: () => {
+      started = false;
+    },
+    refresh: jest.fn(),
     helper: mainHelper, // @TODO: use the Helper from the index once the RoutingManger uses the index
     templatesConfig: {},
     insightsClient: null,
-    scheduleStalledRender: jest.fn(),
-    scheduleSearch: jest.fn(),
-    scheduleRender: jest.fn(),
+    scheduleStalledRender: defer(jest.fn()),
+    scheduleSearch: defer(jest.fn()),
+    scheduleRender: defer(jest.fn()),
     _isSearchStalled: true,
+    _stalledSearchDelay: 200,
+    _searchStalledTimer: null,
     _searchParameters: {},
+    _createURL: jest.fn(() => '#'),
     _createAbsoluteURL: jest.fn(() => '#'),
+    createURL: jest.fn(() => '#'),
+    routing: null, // @TODO: good definition
+    addWidget: jest.fn(),
+    addWidgets: jest.fn(),
+    removeWidget: jest.fn(),
+    removeWidgets: jest.fn(),
+    // @TODO: can be inherited from EventEmitter?
```

Comment:
I tried spreading out `new EventEmitter()`, but that didn't work for TS

Reply:
We want to assign all values to `jest.fn()` anyway so I guess it's fine like this?